src/Polytopes/Cone/constructors.jl

```diff
@@ -33,26 +33,15 @@ julia> HS = Cone(R, L)
 A polyhedral cone in ambient dimension 2
 ```
 """
-struct Cone #a real polymake polyhedron
-    pm_cone::Polymake.BigObject
-end
-function Cone(Rays::Union{Oscar.MatElem,AbstractMatrix})
-    Cone(Polymake.polytope.Cone{Polymake.Rational}(
-        INPUT_RAYS = matrix_for_polymake(Rays),
-    ))
-end
-function Cone(Rays::Union{Oscar.MatElem,AbstractMatrix}, LS::Union{Oscar.MatElem,AbstractMatrix}; non_redundant::Bool=false)
-   if non_redundant
-       Cone(Polymake.polytope.Cone{Polymake.Rational}(
-           RAYS = matrix_for_polymake(Rays),
-           LINEALITY_SPACE = matrix_for_polymake(LS),
-       ))
-   else
-       Cone(Polymake.polytope.Cone{Polymake.Rational}(
-           INPUT_RAYS = matrix_for_polymake(Rays),
-           INPUT_LINEALITY = matrix_for_polymake(LS),
-       ))
-   end
+function Cone(R::Union{PointIterator{Ray}, Oscar.MatElem, AbstractMatrix}, L::Union{PointIterator{Ray}, Oscar.MatElem, AbstractMatrix, Nothing} = nothing; non_redundant::Bool = false)
+    RM = R isa PointIterator{Ray} ? R.m : R isa Union{AnyVecOrMat, Oscar.MatElem} ? matrix_for_polymake(R) : Polymake.Matrix{Polymake.Rational}(undef, 0, size(V.m, 2))
+    LM = L isa PointIterator ? L.m : L isa Union{AnyVecOrMat, Oscar.MatElem} ? matrix_for_polymake(L) : Polymake.Matrix{Polymake.Rational}(undef, 0, size(V.m, 2))
```

Comment:
Why is the code for `RM` checking for `PointsIterator{Ray}` but that for `LM` is just checking for `PointsIterator`? In the signature of the function, both are restricted to `PointsIterator{Ray}`.

Except for this minor difference, these two lines are identical and also identical to similar lines elsewhere in this file. Which would strongly suggest to me to add a helper function for dealing with them. And that could then even use multi dispatch. Say:
```julia
_for_polymake(L::PointsIterator) = L.m
_for_polymake(L::AnyVecOrMat) = matrix_for_polymake(L)
_for_polymake(L::Oscar.MatElem) = matrix_for_polymake(L)
_for_polymake(L::Any) = Polymake.Matrix{Polymake.Rational}(undef, 0, size(L.m, 2) # FIXME is this right? What is it for?
```

Reply:
1. you are right, this is inconsistent at the moment. This is because lineality actually is neither a point nor a ray, so for the moment i decided to go by `Ray` because of the leading zeros of the homogeneous matrix returned by `LINEALITY_SPACE` on a `Polytope` within polymake. I am happy to hear opinions on how I should handle lineality in this context.

2. this idea sounds good. probably directly define `matrix_for_polymake(iter::PointsIterator) = iter.m`?

3. the `FIXME` part covers the case that `nothing` is given because the other types eligible by the signature have been checked out before. it does not suffice to just create any empty matrix because the horizontal size still has to be consistent within the other data, i.e. the dimensions of the vectors of vertices, rays and lineality space generators have to match, even if there are no rays or lineality space generators. so, if `R = nothing` we have `RM` as a `0xn` matrix where `n = size(LM, 2)`.